contracts/locker/BaseLocker.sol

```diff
@@ -75,6 +83,47 @@ contract BaseLocker is
         _setApprovalForAll(address(this), _staking, true);
     }
 
+    /**
+     * @notice Sets the underlying token address
+     * @param _token The new token address
+     */
+    function setTokenAddress(address _token) external onlyOwner {
+        address oldToken = address(underlying);
+        underlying = IERC20(_token);
+        emit TokenAddressSet(oldToken, _token);
+    }
+
+    /**
+     * @notice Sets the staking contract address
+     * @param _staking The new staking contract address
+     */
+    function setStakingAddress(address _staking) external onlyOwner{
+        address oldStaking = address(staking);
+        staking = IOmnichainStaking(_staking);
+        emit StakingAddressSet(oldStaking, _staking);
+    }
+
+    /**
+     * @notice Sets the staking bonus contract address
+     * @param _stakingBonus The new staking bonus contract address
+     */
+    function setStakingBonusAddress(address _stakingBonus) external onlyOwner {
+        address oldStakingBonus = address(0); // Update this with the actual old staking bonus address if needed
+        _setApprovalForAll(address(this), oldStakingBonus, false);
+        _setApprovalForAll(address(this), _stakingBonus, true);
+        emit StakingBonusAddressSet(oldStakingBonus, _stakingBonus);
+    }
+
+    /**
+     * @notice Sets the maximum lock time
+     * @param _maxTime The new maximum lock time in seconds
+     */
+    function setMaxTime(uint256 _maxTime) external onlyOwner {
```

Comment:
We wouldn't need to change `MAXTIME` ever because that'd screw up all the previous locks. So this function would be redundant.

Reply:
Okay, Removed this function